src/cgutils.cpp

```diff
@@ -921,6 +921,8 @@ JL_DLLEXPORT Type *julia_type_to_llvm(jl_value_t *jt, bool *isboxed)
         return PointerType::get(lt, 0);
     }
     if (jl_is_bitstype(jt)) {
+        if (jt == (jl_value_t*)jl_long_type)
+            return T_size;
```

Comment:
we could probably also just generally generically cache these in `jst->struct_decl`